docs/recipes/storage/use-sqlprovider-ssdt.md

```diff
@@ -0,0 +1,230 @@
+
+![image](../../img/sql-provider5.png)
+
+5) Finally, right click the SafeTodoDB project and select "Build". This will create a .dacpac file which we will use in the next step.
+
+
+## Create a TodoRepository Using the new SSDT provider in SQLProvider
+
+### Installing SQLProvider from NuGet
+
+Install dependencies `SqlProvider` and `Microsoft.Data.SqlClient`
+
+```
+dotnet paket add SqlProvider -p Server
+dotnet paket add Microsoft.Data.SqlClient -p Server
+```
+
+### Initialize Type Provider
+Next, we will wire up our type provider to generate database types based on the compiled .dacpac file.
+
+1) In the Server project, create a new file, `Database.fs`. (this should be above `Server.fs`).
+
+```fsharp
+module Database
+open FSharp.Data.Sql
+
+[<Literal>]
+let SsdtPath = __SOURCE_DIRECTORY__ + @"/../../ssdt/SafeTodoDB/bin/Debug/SafeTodoDB.dacpac"
+
+// TO RELOAD SCHEMA: 1) uncomment the line below; 2) save; 3) recomment; 4) save again and wait.
+//DB.GetDataContext().``Design Time Commands``.ClearDatabaseSchemaCache
+
+type DB = 
+    SqlDataProvider<
+        Common.DatabaseProviderTypes.MSSQLSERVER_SSDT,
+        SsdtPath = SsdtPath,
+        UseOptionTypes = Common.NullableColumnType.OPTION
+    >
+
+let createContext (connectionString: string) =
+    DB.GetDataContext(connectionString)
+```
+
+2) Create `TodoRepository.fs` below `Database.fs`.
+
+``` fsharp
+module TodoRepository
+open FSharp.Data.Sql
+open Database
+open Shared
+
+/// Get all todos that have not been marked as "done". 
+let getTodos (db: DB.dataContext) = 
+    query {
+        for todo in db.Dbo.Todos do
+        where (not todo.IsDone)
+        select 
+            { Shared.Todo.Id = todo.Id
+              Shared.Todo.Description = todo.Description }
+    }
+    |> List.executeQueryAsync
+
+let addTodo (db: DB.dataContext) (todo: Shared.Todo) =
+    async {
+        let t = db.Dbo.Todos.Create()
+        t.Id <- todo.Id
+        t.Description <- todo.Description
+        t.IsDone <- false
+
+        do! db.SubmitUpdatesAsync() |> Async.AwaitTask
+    }
+```
+
+3) Create `TodoController.fs` below `TodoRepository.fs`.
+
+``` fsharp
+module TodoController
+open Database
+open Shared
+
+let getTodos (db: DB.dataContext) = 
+    TodoRepository.getTodos db |> Async.AwaitTask
+
+let addTodo (db: DB.dataContext) (todo: Todo) = 
+    async {
+        if Todo.isValid todo.Description then
+            do! TodoRepository.addTodo db todo
+            return todo
+        else 
+            return failwith "Invalid todo"
+    }
+```
+
+4) Finally, replace the stubbed todosApi implementation in `Server.fs` with our type provided implementation.
+
+``` fsharp
+module Server
+
+open Fable.Remoting.Server
+open Fable.Remoting.Giraffe
+open Saturn
+open System
+open Shared
+open Microsoft.AspNetCore.Http
+
+let todosApi =
+    let db = Database.createContext @"Data =.\SQLEXPRESS,1433;Database=SafeTodo;User ID=sa;Password=yourStrong(!)Password;Encrypt=false"
```

Comment:
Could not get this to work with the original Connection string  "Data Source=.\SQLEXPRESS;Initial Catalog=SafeTodo;Integrated Security=SSPI;". I suppose that one could be used for both a prod and local environment? I feel like adding instructions on how to have separate connection strings for prod and local might be useful anyway. Should we get the old connection string to work, or should we add instructions on how to add separate connection strings?